scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala

```diff
@@ -309,16 +308,21 @@ class RedundantBraces(ftoks: FormatTokens) extends FormatTokensRewrite.Rule {
             case Type.Name("Unit") => true
             case _ => false
           }
-        settings.defnBodies &&
         checkBlockAsBody(b, d.body) &&
+        defnBodies(d.name, d.tparams, d.paramss) &&
         !isProcedureSyntax(d) &&
         !disqualifiedByUnit
 
-      case d: Defn.Var => d.rhs.exists(checkBlockAsBody(b, _))
-      case d: Defn.Val => checkBlockAsBody(b, d.rhs)
-      case d: Defn.Type => checkBlockAsBody(b, d.body)
-      case d: Defn.Macro => checkBlockAsBody(b, d.body)
-      case d: Defn.GivenAlias => checkBlockAsBody(b, d.body)
+      case d: Defn.Var =>
+        defnBodies(true) && d.rhs.exists(checkBlockAsBody(b, _))
```

Comment:
```suggestion
        defnBodies(noParams = true) && d.rhs.exists(checkBlockAsBody(b, _))
```
Otherwise it's a bit unclear

Reply:
will do